stellar-contract-env-common/src/env.rs

```diff
@@ -151,6 +154,29 @@ macro_rules! call_macro_with_all_host_functions {
                 {"$L", fn bigint_to_i64(x:Object) -> i64}
                 {"$M", fn bigint_from_i64(x:i64) -> Object}
             }
+
+            mod binary "x" {
+                {"$_", fn serialize_to_binary(x:Object) -> Object}
```

Comment:
❓ Do the serialization functions serialize to an ScVal? Or to the raw type?

For example, does an i32 serialize as an XDR integer, or as an XDR integer within an ScVal union with the I32 discriminant? I assume it's the latter, but unclear.

Reply:
Actually, both the serialize and the deserialize functions are intended to operate on `SCObject`s (and the function argument type `Object` reflects that). My rationale is all other non-object `SCVal` types can be trivially converted between bytes and an i64. Do you feel we should expose this to take in an `SCVal`?